force-app/main/default/classes/Homework.cls

```diff
@@ -51,13 +83,38 @@ public with sharing class Homework
     //    position of all occurrences.
     public Map<String, List<Integer>> getPositionsByCharacter(String value)
     {
+        Map<String, List<Integer>> posMap = new Map<String, List<Integer>>();
+        List<String> valueList = value.split('');
+        Set<String> uniqueChars = new Set<String>(valueList);
+        List<String> charsToIterate = new List<String>(uniqueChars);
+        
+        for (String c : charsToIterate) {
+            Integer pos = 0;
+            List<Integer> foundPositions = new List<Integer>();
+            
+            for (String v : valueList) {
+                if(v.toLowerCase() == c.toLowerCase())
+                {
+                    foundPositions.add(pos);
+                }
+                pos++;
+            }
```

Comment:
This will work fine, but nesting loops can become expensive in terms of CPU time consumed when iterating over large collections. In my view it would be better to iterate `valueList` and create/append to the list of matches in the map. Here's my solution for reference:

```
Map<String, List<Integer>> positionsByCharacter=new Map<String, List<Integer>>();

for (Integer idx=0; idx<value.length(); idx++)
{
    String charVal=value.substring(idx, idx+1);
    List<Integer> positions=positionsByCharacter.get(charVal);
    if (null==positions) 
    {
        positions=new List<Integer>();
        positionsByCharacter.put(charVal, positions);
    }
    positions.add(idx)
}
```